### PR TITLE
Fix: #24023 - Ensure different language in translation split view

### LINKED ISSUE #24023: Translation Split-View always opens with the same language
### Describe the Bug

When selecting the split view for translations, the second view is intended to open with an alternative language. However, the current behavior is that both views are always open with the base language rather than displaying the translation.

issue

### To Reproduce

1. Ensure at least 2 languages are configured in the system, such as `en` and `fi`.
2. Within a collection, set a `translation` field.
3. Configure the field to use the `Split-View` mode.
4. When accessing the `translation` field, the expectation is that the second view should display the alternative language (e.g. `fi`), but the current behavior is that both views show the base language (`en`).

reproduce

### Directus Version

v11.2.1

### Hosting Strategy

Self-Hosted (Docker Image)

### Database

PostgreSQL 15

### PR DESCRIPTION
**Pull request by eremannisto**
## Scope

What's changed:

- Modified language selection logic in split-view to prevent duplicate languages
- Improved TypeScript typing for null safety
- Added better fallback logic for language selection

**Before, would open with the same language:**
before

**All languages added to the `languages` collection:**
languages

**Default langauge is set to English:**
english

**Default language is set to Finnish:**
finnish

**No default language selected:**
none

**Only one language set in the `languages` collection:**
one

## Potential Risks / Drawbacks

- None identified - changes only affect the UI behavior of translation split-view
- Maintains existing fallback behavior if only one language is available

## Review Notes / Questions

- The fix prioritizes showing different languages in split-view while respecting user/default language preferences
- The changes include TypeScript improvements for better type safety

---

Fixes [#24023](https://github.com/directus/directus/issues/24023)

### CODE REVIEW

**Comment by eremannisto** [comment]
Hi! First time contributing to Directus here. Please let me know if any adjustments are needed to align with the project's PR guidelines.

**Review by AlexGaillard — DISMISSED** [review]
Thanks for the PR @eremannisto 👋 

Great work finding the source of the problem and also the fix. I've included some thoughts about how we can maybe simplify this a little bit and maybe be a bit more performant.

I think we can get this wrapped up pretty quickly 🚀

File: app/src/interfaces/translations/translations.vue
Comment (AlexGaillard) [reply app/src/interfaces/translations/translations.vue:251]: I think this comment got doubled when github conked out on me during review 🙃

**Comment by eremannisto** [comment]
Thanks @AlexGaillard for the detailed code review and optimization suggestions! I made the suggested changes and tested it, and seems to work perfeclty. :)

**Review by rijkvanzanten — APPROVED** [review]

**Comment by rijkvanzanten** [comment]
Thanks @eremannisto! And welcome to the repo 🙂 I'll let @AlexGaillard smack that merge button. LGTM! 🚀

**Pull request merged**
